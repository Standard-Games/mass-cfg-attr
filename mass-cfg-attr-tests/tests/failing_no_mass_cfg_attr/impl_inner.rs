#![deny(unused_must_use, deprecated)]
#![allow(path_statements)] // disable warning for test

struct Struct();

impl Struct {
    #![deprecated]

    #[must_use]
    pub fn get_random_number() -> u8 {
        4 // chosen by fair dice roll
    }
}

fn main() {
    // Violation of unused_must_use if not removed by mass_cfg_attr
    Struct::get_random_number();
}
