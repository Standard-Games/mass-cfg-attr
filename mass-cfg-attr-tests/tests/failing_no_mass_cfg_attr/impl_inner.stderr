error: use of deprecated associated function `Struct::get_random_number`
  --> tests/failing_no_mass_cfg_attr/impl_inner.rs:17:13
   |
17 |     Struct::get_random_number();
   |             ^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> tests/failing_no_mass_cfg_attr/impl_inner.rs:1:26
   |
1  | #![deny(unused_must_use, deprecated)]
   |                          ^^^^^^^^^^

error: unused return value of `Struct::get_random_number` that must be used
  --> tests/failing_no_mass_cfg_attr/impl_inner.rs:17:5
   |
17 |     Struct::get_random_number();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> tests/failing_no_mass_cfg_attr/impl_inner.rs:1:9
   |
1  | #![deny(unused_must_use, deprecated)]
   |         ^^^^^^^^^^^^^^^
